traffic_ops/testing/api/v4/parameters_test.go

```diff
@@ -231,3 +340,204 @@ func DeleteTestParameter(t *testing.T, pl tc.Parameter) {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+		if len(resp) <= 0 {
+			t.Errorf("No data available for Get Parameters by Config file")
+		}
+	}
+}
+
+func GetTestParametersByValue(t *testing.T) {
+	for _, parameters := range testData.Parameters {
+		_, reqInf, err := TOSession.GetParametersByValue(parameters.Value, nil)
+		if err != nil {
+			t.Errorf("cannot GET Parameter by Value: %v - %v", parameters.Value, err)
+		}
+		if reqInf.StatusCode != http.StatusOK {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
+
+func GetTestParametersByName(t *testing.T) {
+	for _, parameters := range testData.Parameters {
+		resp, reqInf, err := TOSession.GetParametersByName(parameters.Name, nil)
+		if err != nil {
+			t.Errorf("cannot GET Parameter by Name: %v - %v", parameters.Name, err)
+		}
+		if reqInf.StatusCode != http.StatusOK {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+		if len(resp) <= 0 {
+			t.Errorf("No data available for Get Parameters by Name")
+		}
+	}
+}
+
+func GetParametersByInvalidId(t *testing.T) {
+	resp, _, err := TOSession.GetParameterByID(10000, nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid ID %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid ID shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidName(t *testing.T) {
+	resp, _, err := TOSession.GetParametersByName("abcd", nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid Name %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid name shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidConfigfile(t *testing.T) {
+	resp, _, err := TOSession.GetParametersByConfigFile("abcd", nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid ConfigFile %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid config file shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidValue(t *testing.T) {
+	resp, _, err := TOSession.GetParametersByValue("abcd", nil)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid value %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Errorf("Invalid value shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func CreateTestParametersAlreadyExist(t *testing.T) {
+	resp, _, _ := TOSession.GetParameters(nil, nil)
+	_, reqInf, _ := TOSession.CreateParameter(resp[0])
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+}
+
+func CreateTestParametersMissingName(t *testing.T) {
+	firstParameter := testData.Parameters[0]
+	firstParameter.Name = ""
+	_, reqInf, _ := TOSession.CreateParameter(firstParameter)
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+}
+
+func CreateTestParametersMissingconfigFile(t *testing.T) {
+	firstParameter := testData.Parameters[0]
```

Comment:
This will segfault if the testing data is in a bad state

Reply:
Added null checks.